youtube_dl/extractor/common.py

```diff
@@ -438,6 +439,10 @@ def _guess_encoding_from_content(content_type, webpage_bytes):
     def _webpage_read_content(self, urlh, url_or_request, video_id, note=None, errnote=None, fatal=True, prefix=None, encoding=None):
         content_type = urlh.headers.get('Content-Type', '')
         webpage_bytes = urlh.read()
+        webpage_bytes, bom_enc = parse_strip_bom(webpage_bytes)
```

Comment:
You are right it is not directly related to the PR. That's what i meant with the above comment about the separate pull. 
The reason i need this is that the site sends unicode POMs which lead to a json parser error. My first attempt (see ed10b1c) was to strip the prefix using  `transform_source` but it turned out to be trickier than i thought. The best i could come up with was the function `strip_bom_utf8`. But apperently this does not work with python 3.2.
This latest change is an attempt to solve the issue in a more general way. As i think i'm not the only one that has to work around json parser errors and unicode POMs. I'm open to suggestions as to how to fix the problem.